parsers/ocaml.c

```diff
@@ -1730,12 +1929,22 @@ static void localScope (vString * const ident, ocaToken what)
 		toDoNext = &contextualTillToken;
 		waitedToken = Tok_CurlR;
 		comeAfter = &localScope;
-		contextualTillToken (ident, what);
+		contextualTillToken (ident, what, whatNext);
 		break;
 
 		/* Yeah imperative feature of OCaml,
 		 * a ';' like in C */
 	case Tok_semi:
+		/* ';;' case should end all scopes */
```

Comment:
if `;;` is a special token (and not just `;` followed by a second `;` -- e.g. that `;;` and `; ;` aren't the same) it should probably rather be recognized as such?  (e.g. emit something like `Tok_souble_semi` or something)